llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -81,6 +81,11 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
         .clampScalar(BigTyIdx, XLenLLT, XLenLLT);
   }
 
+  getActionDefinitionsBuilder(G_BSWAP)
+    .widenScalarToNextPow2(0)
```

Comment:
why is the widenScalar and clampScalar needed? It doens't look like they are being tested.

Reply:
widenScalarToNextPow2 I grabbed from AArch64 who does not seem to be testing it as well. I agree that we can drop it because the intrinsic documentation says it will always be a power of 2. The clampScalar I think is needed in the case that there is a bswap.i64 on rv32? The last commit shows the impact on the diff. Do you think it was correct before the last commit?

Reply:
> Any plans to limit the legal types?

@tschuett did you have something else in mind?

Reply:
As far as I know bswap isn't required to be a power of 2. Just multiple of 16 bits.

Reply:
I guess the clamp causes it to be split as a BSWAP instead of creating a bunch of i64 shifts, ands, ors that need to be independently split?

Reply:
I thought of `legalFor`, but ...

Reply:
My understanding is that `legalFor` would mark those types as legal and not need to lower them. At least that is what it looks like when I try it. Am I misunderstanding something?

Reply:
No. You have currently no instructions, thus you must clamp and lower.